src/csl/mod.rs

```diff
@@ -2852,4 +2866,72 @@ mod tests {
             // }
         }
     }
+
+    #[test]
+    fn test_citation_collapsing() {
```

Comment:
Thanks to PR #168, we now have local citeproc-style unit tests at `tests/local`. Do you think you could create a test in `tests/local` instead? That should allow us to remove both this function and the `styles/numeric.csl` file (see the existing test there for a sample; see https://github.com/citation-style-language/test-suite for more information on the test format).